### PR TITLE
depends: systemtap 4.8

### LINKED ISSUE #26916: tracepoints: gnu-zero-variadic-macro-arguments warnings
These occur for all tracepoints when using Clang 15 and a system installed `systemtap`:
```bash
# clang --version
Debian clang version 15.0.7

./autogen.sh
./configure CC=clang-15 CXX=clang++-15
make
...
net_processing.cpp:4892:5: warning: must specify at least one argument for '...' parameter of variadic macro [-Wgnu-zero-variadic-macro-arguments]
    TRACE6(net, inbound_message,
    ^
./util/trace.h:18:50: note: expanded from macro 'TRACE6'
#define TRACE6(context, event, a, b, c, d, e, f) DTRACE_PROBE6(context, event, a, b, c, d, e, f)
                                                 ^
/usr/include/aarch64-linux-gnu/sys/sdt.h:494:3: note: expanded from macro 'DTRACE_PROBE6'
  STAP_PROBE6(provider,probe,parm1,parm2,parm3,parm4,parm5,parm6)
  ^
/usr/include/aarch64-linux-gnu/sys/sdt.h:380:3: note: expanded from macro 'STAP_PROBE6'
  _SDT_PROBE(provider, name, 6, (arg1, arg2, arg3, arg4, arg5, arg6))
  ^
/usr/include/aarch64-linux-gnu/sys/sdt.h:78:75: note: expanded from macro '_SDT_PROBE'
    __asm__ __volatile__ (_SDT_ASM_BODY(provider, name, _SDT_ASM_ARGS, (n)) \
                                                                          ^
/usr/include/aarch64-linux-gnu/sys/sdt.h:283:9: note: macro '_SDT_ASM_BODY' defined here
#define _SDT_ASM_BODY(provider, name, pack_args, args, ...)                   \
        ^
1 warning generated.
```

### PR DESCRIPTION
**Pull request by fanquake**
Updates systemtap to 4.8.
Includes acc2895a66a4b654e9a0a05ed0927f67f48c75b2 from #25972.
Will half (depends) fix #26916.

Release notes etc: https://lwn.net/Articles/913908/.

### CODE REVIEW

**Comment by 0xB10C** [comment]
ACK a91379359daca24b83b5e98714ec2fda93f24185

Changes to the depends look good. Hash matches with what I get.

 > Release notes etc: https://lwn.net/Articles/913908/.

[sourceware.org/PR29246](https://sourceware.org/bugzilla/show_bug.cgi?id=29246) is relevant to us which is the only [commit](https://sourceware.org/git/?p=systemtap.git;a=commitdiff;h=97213111b28685bf5f3edad5ec9e8bca397e8a5f;hp=6f6f15a002da7a05e64ef24047399688809895b2) changing the `includes/sys/sdt.h` file. But it shouldn't affect our use of sdt.h.

I've [previously checked](https://github.com/bitcoin/bitcoin/pull/25972#issuecomment-1246700093) that removing the [problematic]() variadic in a patch does not cause any problems for us as we don't use it. This also doesn't conflict with https://github.com/bitcoin/bitcoin/pull/26593 as we use a different variadic there.

**Comment by hebasto** [comment]
> Updates systemtap to 4.8.
> Includes [acc2895](https://github.com/bitcoin/bitcoin/commit/acc2895a66a4b654e9a0a05ed0927f67f48c75b2) from #25972.
> Will half (depends) fix #26916.

Not able to reproduce #26916 when building with depends on the master branch on Ubuntu 23.10:
```
$ make -C depends NO_QT=1 NO_WALLET=1 NO_ZMQ=1 NO_UPNP=1 NO_NATPMP=1
$ ./autogen.sh
$ ./configure CONFIG_SITE=$PWD/depends/x86_64-pc-linux-gnu/share/config.site CC=clang CXX=clang++
$ make clean
$ make
$ clang --version
Ubuntu clang version 15.0.2-1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

Is it a result of bumping to 4.8?

**Comment by fanquake** [comment]
> Is it a result of bumping

No it's probably because you don't have any warnings turned on.

**Comment by hebasto** [comment]
> > Is it a result of bumping
> 
> No it's probably because you don't have any warnings turned on.

Right... :man_facepalming:

File: depends/patches/systemtap/fix_variadic_warning.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/includes/sys/sdt.h b/includes/sys/sdt.h
+index 4075a5f..7c6138c 100644
+--- a/includes/sys/sdt.h
++++ b/includes/sys/sdt.h
+@@ -278,7 +278,7 @@ __extension__ extern unsigned long long __sdt_unsp;
```
Comment (hebasto) [inline depends/patches/systemtap/fix_variadic_warning.patch:5]:
```suggestion
@@ -276,7 +276,7 @@ __extension__ extern unsigned long long __sdt_unsp;
```

File: depends/patches/systemtap/fix_variadic_warning.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+   _SDT_ASM_1(.purgem _SDT_TYPE_)					      \
+   _SDT_ASM_1(.purgem _SDT_TYPE)
+ 
+-#define _SDT_ASM_BODY(provider, name, pack_args, args, ...)		      \
++#define _SDT_ASM_BODY(provider, name, pack_args, args)		            \
```
Comment (hebasto) [inline depends/patches/systemtap/fix_variadic_warning.patch:10]: Add a comment that this patch can be dropped after switching to [C++20](https://github.com/bitcoin/bitcoin/issues/26916#issuecomment-1400239451)?

**Review by hebasto — APPROVED** [review]
ACK a91379359daca24b83b5e98714ec2fda93f24185, tested on Ubuntu 22.10.

File: depends/patches/systemtap/fix_variadic_warning.patch (hunk 2 above)
Comment (fanquake) [reply depends/patches/systemtap/fix_variadic_warning.patch:13]: Added.

**Review by hebasto — APPROVED** [review]
re-ACK df7ae8b7ca9cf53357d6a1c6bb70a2e40d2a1017.

**Comment by 0xB10C** [comment]
ACK df7ae8b7ca9cf53357d6a1c6bb70a2e40d2a1017

**Pull request merged**